### PR TITLE
Release 2.0.0a3

### PR DESCRIPTION
**Pull request by pquentin**
* [x]  See if all tests, including integration, pass
* [ ]  Get the release pull request approved by a [CODEOWNER](https://github.com/urllib3/urllib3/blob/main/.github/CODEOWNERS)
* [ ]  Squash merge the release pull request with message "`Release <VERSION>`"
* [ ]  Tag with X.Y.Z, push tag on urllib3/urllib3 (not on your fork, update `<REMOTE>` accordingly)
  * Notice that the `<VERSION>` shouldn't have a `v` prefix (Use `1.26.6` instead of `v.1.26.6`)
  * ```
    git tag -s -a '<VERSION>' -m 'Release: <VERSION>'
    git push <REMOTE> --tags
    ```
* [ ]  Execute the `publish` GitHub workflow. This requires a review from a maintainer.
* [ ]  Ensure that all expected artifacts are added to the new GitHub release. Should
       be one `.whl`, one `.tar.gz`, and one `urllib3.intoto.jsonl`. Update the GitHub
       release to have the content of the release's changelog.
* [ ]  Announce on:
  * [ ]  Twitter
  * [ ]  Discord
  * [ ]  OpenCollective
* [ ]  Update Tidelift metadata
* [ ]  If this was a 1.26.x release, add changelog to the `main` branch

### CODE REVIEW

**Review by sethmlarson — DISMISSED** [review]
LGTM!

File: CHANGES.rst [hunk 1]
```diff
@@ -1,4 +1,15 @@
-2.0.0a2 (2022-11-23)
+2.0.0a2 (2023-01-07)
```
Comment (zhanpon) [inline CHANGES.rst:1]:
```suggestion
2.0.0a3 (2023-01-07)
```

File: CHANGES.rst [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+
+Bugfixes
+--------
+
+- Fix logging error when using ``add_stderr_logger``. (`#2839 <https://github.com/urllib3/urllib3/issues/2839>`__)
+- Fixed parsing of port 0 (zero) returning None, instead of 0. (`#2850 <https://github.com/urllib3/urllib3/issues/2850>`__)
+- Fixed the type hint of ``PoolKey.key_retries`` by adding ``bool`` to the union. (`#2865 <https://github.com/urllib3/urllib3/issues/2865>`__)
+
+
+2.0.0a3 (2022-11-23)
```
Comment (zhanpon) [inline CHANGES.rst:12]:
```suggestion
2.0.0a2 (2022-11-23)
```

**Review by zhanpon — COMMENTED** [review]
Just a little typo 😄

**Comment by sethmlarson** [comment]
Great catch @zhanpon, thank you.

**Review by sethmlarson — APPROVED** [review]
:shipit:

**Pull request merged**